SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java

```diff
@@ -312,6 +313,20 @@ private boolean requestTypeIsOnDemand(SingularityPendingTaskId taskId) {
     return false;
   }
 
+  private List<SingularityPendingTaskId> getOnDemandLateTasks (List<SingularityPendingTaskId> maybeOnDemandLateTasks) {
+    List<SingularityPendingTaskId> onDemandLateTasks = new ArrayList<>();
+    for (SingularityPendingTaskId maybeOnDemandLateTask : maybeOnDemandLateTasks) {
+      String requestId = requestManager.getRequest(maybeOnDemandLateTask.getRequestId()).get().getRequest().getId();
```

Comment:
we don't need to fetch this here, requestId is already on the SingularityPendingTaskId, and you are fetching the SingularityRequest on the next line down